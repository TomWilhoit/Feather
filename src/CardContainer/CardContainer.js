import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from 'react-router-dom';
import Card from "../Card/Card";
import PropTypes from "prop-types";
import "./CardContainer.scss";
import  CardDetail  from "../CardDetail/CardDetail";

export class CardContainer extends Component {
  constructor() {
    super();
    this.state = {
      displayDetail: false,
      currentCard: null
    }
  }

  returnToContainer = () => {
    this.setState({
      displayDetail: false
    })
  }

  toggleDisplay = (e) => {
    this.setState({
        displayDetail: true,
        currentCard: e.target.parentElement.attributes.value.value 
      })
  }

  render() {
    if(this.state.displayDetail === false){
      return (
        <div className="card-cont">
          {this.props.locations.map((location, index) => (
          <Link key={index} to={`/${location.id}`}>
        <Card  toggleDisplay={this.toggleDisplay}  key={index} index={index} place={location} />
          </Link>
      ))}
        </div>
      );
    }else{
      return(
        <div>
          
          <CardDetail returnToContainer={this.returnToContainer} singleCard={this.state.currentCard}/>
        </div>
      )
    }
  }
}

CardContainer.propTypes = {
  locations: PropTypes.array,
  weather: PropTypes.array
};

export const mapStateToProps = state => ({
  locations: state.locations,
  weather: state.weather
});

export default connect(
  mapStateToProps,
  null
)(CardContainer);
