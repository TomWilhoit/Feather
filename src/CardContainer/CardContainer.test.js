import React from "react";
import { CardContainer } from "./CardContainer";
import { mapStateToProps } from "./CardContainer";
import { shallow } from "enzyme";

describe("CardContainer", () => {
  let wrapper;
  let mockProps = {
    locations: ["Chattanooga", "Memphis"]
  };
  beforeEach(() => {
    wrapper = shallow(<CardContainer locations={mockProps.locations} />);
  });

  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have default state", () => {
    expect(wrapper.state()).toEqual({
      currentCard: null,
      displayDetail: false
    });
  });

  it("should change the state of displayDetail when returnToContainer is invoked", () => {
    const mockEvent = {target:{ parentElement : { attributes: {value:{value: 0}}}}}
    wrapper.setState({ displayDetail: true });
    wrapper.instance().toggleDisplay(mockEvent);
    expect(wrapper.state()).toEqual({
      currentCard: 0,
      displayDetail: true
    });
  });

  it("should change the state of displayDetail when toggleDisplay is invoked", () => {
    wrapper.setState({ displayDetail: true });
    wrapper.instance().returnToContainer();
    expect(wrapper.state()).toEqual({
      currentCard: null,
      displayDetail: false
    });
  });

  describe("mapStateToProps", () => {
    it("should mapStateToProps", () => {
      const mockState = {
        locations: ["Chattanooga", "Memphis"]
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockState);
    });
  });
});
