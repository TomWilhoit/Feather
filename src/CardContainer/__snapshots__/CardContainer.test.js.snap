// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot with all data passed in 1`] = `
<div
  className="card-cont"
>
  <Link
    key="0"
    to="/undefined"
  >
    <ConnectFunction
      index={0}
      key="0"
      place="Chattanooga"
      toggleDisplay={[Function]}
    />
  </Link>
  <Link
    key="1"
    to="/undefined"
  >
    <ConnectFunction
      index={1}
      key="1"
      place="Memphis"
      toggleDisplay={[Function]}
    />
  </Link>
</div>
`;
